File: tests/src/test_entity.cpp
#include <gtest/gtest.h>

#include <optional>
#include <string>

#include "entity.hpp"

TEST(EntityTest, ConstructorWithNameAndOptionalKeyAndIntValue) {
  std::string name = "Entity";
  std::optional<std::string> key = "KEY";
  int value = 42;
  int padding = 2;

  Entity entity(name, key, value, padding);

  EXPECT_EQ(entity.id(), "KEY-42");
}

TEST(EntityTest, ConstructorWithNameAndOptionalKeyAndStringValue) {
  std::string name = "Entity";
  std::optional<std::string> key = "KEY";
  std::string value = "VALUE";
  int padding = 2;

  Entity entity(name, key, value, padding);

  EXPECT_EQ(entity.id(), "KEY-VALUE");
}

TEST(EntityTest, ConstructorWithNameAndOptionalKeyAndInvalidStringValue) {
  std::string name = "Entity";
  std::optional<std::string> key = "KEY";
  std::string value = "INVALID";
  int padding = 2;

  Entity entity(name, key, value, padding);

  EXPECT_EQ(entity.id(), "KEY-INVALID");
}

File: tests/test_entity.py
from typing import Optional

from libbids.clibbids import Entity  # type: ignore


def test_constructor_with_name_and_optional_key_and_int_value() -> None:
    name = "Entity"
    key: Optional[str] = "KEY"
    value: int = 42
    padding: int = 2

    entity = Entity(name, key, value, padding)

    assert entity.id == "KEY-42"


def test_constructor_with_name_and_optional_key_and_string_value() -> None:
    name = "Entity"
    key: Optional[str] = "KEY"
    value: str = "VALUE"
    padding: int = 2

    entity = Entity(name, key, value, padding)

    assert entity.id == "KEY-VALUE"


def test_constructor_with_name_and_optional_key_and_invalid_string_value() -> None:
    name: str = "Entity"
    key: Optional[str] = "KEY"
    value: str = "INVALID"
    padding: int = 2

    entity: Entity = Entity(name, key, value, padding)

    assert entity.id == "KEY-INVALID"
    assert entity.padding == 2
